plugin/PDFDocumentPage.qml

```diff
@@ -144,6 +217,113 @@ DocumentPage {
+                        textTool.first = false
+                    }
+                }
+                IconButton {
+                    id: textButton
+                    anchors.centerIn: parent
+                    highlighted: pressed || textTool.pressed || row.activeItem === textTool
+                    icon.source: "image://theme/icon-m-notifications"
+                    onClicked: textTool.clicked(mouse)
+                }
+                MouseArea {
+                    parent: row.activeItem === textTool ? view : null
+                    anchors.fill: parent
+                    onClicked: {
+                        var annotation = textComponent.createObject(contextMenuText)
+                        var pt = Qt.point(view.contentX + mouse.x,
+                                          view.contentY + mouse.y)
+                        pdfDocument.create(annotation,
+                                           function() {
+                                               var at = view.getPositionAt(pt)
+                                               annotation.attachAt(pdfDocument,
+                                                                   at[0], at[2], at[1])
+                                           })
+                        row.toggle(textTool)
+                    }
+                }
+            }
+            BackgroundItem {
+                id: highlightTool
+                property bool first: true
+
+                function highlightSelection() {
+                    var anno = highlightComponent.createObject(highlightTool)
+                    anno.color = highlightColorConfig.value
+                    anno.style = highlightStyleConfig.toEnum(highlightStyleConfig.value)
+                    anno.attach(pdfDocument, view.selection)
+                    toolbar.hide()
+                }
+
+                width: row.itemWidth
+                height: parent.height
+                highlighted: pressed || highlightButton.pressed
+                onClicked: {
+                    if (view.selection.selected) {
+                        highlightSelection()
+                        view.selection.unselect()
+                        return
+                    }
+                    row.toggle(highlightTool)
+                    if (highlightTool.first) {
+                        //% "Tap and move your finger over the area"
+                        notice.show(qsTrId("sailfish-office-la-notice-anno-highlight"))
+                        highlightTool.first = false
+                    }
+                }
+                IconButton {
+                    id: highlightButton
+                    anchors.centerIn: parent
+                    highlighted: pressed || highlightTool.pressed || row.activeItem === highlightTool
+                    icon.source: "image://theme/icon-m-edit"
+                    onClicked: highlightTool.clicked(mouse)
+                }
+                Component {
```

Comment:
Ok, moved it first because it's called from the function that is on top of this BackgroundItem.